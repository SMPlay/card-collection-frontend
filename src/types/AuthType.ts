export interface AuthType<T> {
  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  handleChange: (e: React.ChangeEvent<any>) => void;
  handleBlur: (e: any) => void;
  values: T;
  status: string;
  errors: T;
  touched: T;
}
