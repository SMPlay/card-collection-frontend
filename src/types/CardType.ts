export interface CardType {
  id: string;
  name: string;
  imageUrl: string;
  role: string;
  rarity: string;
  need: number;
  have: number;
  number: number;
}