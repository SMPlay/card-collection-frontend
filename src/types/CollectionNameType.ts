export interface CollectionNameType {
  id: string;
  name: string;
}
