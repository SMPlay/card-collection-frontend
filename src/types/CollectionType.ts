export interface CollectionType {
  name: string;
  imageUrl: string;
  id: string;
  cardsCount: number;
  release: number;
}
