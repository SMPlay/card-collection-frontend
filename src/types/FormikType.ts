interface Status {
  loading: boolean;
  error: string;
}

export interface FormikType<T> {
  handleSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  handleChange: (e: React.ChangeEvent<any>) => void;
  values: T;
  status: Status;
  errors: T;
}
