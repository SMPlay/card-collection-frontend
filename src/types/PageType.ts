export interface PageType {
  pageName: string;
  url: string;
}
