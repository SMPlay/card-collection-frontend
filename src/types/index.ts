export { CollectionType } from './CollectionType'
export { PageType } from './PageType'