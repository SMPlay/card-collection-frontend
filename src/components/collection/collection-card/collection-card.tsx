import React, { FC } from "react";
import makeStyles from "@material-ui/core/styles/makeStyles";
import { Card, CardActionArea, CardContent, CardMedia, createStyles, Theme, Typography } from "@material-ui/core";

import { CardType } from "../../../types/CardType";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    metaText: {
      padding: "0 15px"
    },
    card: {
      height: 550,
    }
  })
);

export const CollectionCard: FC<CardType> = ({imageUrl, rarity, name, number, role, need, have}) => {
  const styles = useStyles();

  return <Card>
    <CardActionArea>
      <CardMedia
        component="img"
        alt={name}
        height="450"
        image={imageUrl}
        title={name}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">
          {name}
        </Typography>
      </CardContent>
    </CardActionArea>
    <CardActionArea className={styles.metaText + " " + styles.card}>
      <Typography variant="h6" component="h5">
        Номер карты: {number}
      </Typography>
      <Typography variant="h6" component="h5">
        Роль: {role}
      </Typography>
      <Typography variant="h6" component="h5">
        Редкость: {rarity}
      </Typography>
      <Typography variant="h6" component="h5">
        Есть карт: {have}
      </Typography>
      <Typography variant="h6" component="h5">
        Нужно карт: {need}
      </Typography>
    </CardActionArea>
  </Card>
}