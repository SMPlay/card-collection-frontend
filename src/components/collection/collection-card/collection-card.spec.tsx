import React from "react";
import { create } from "react-test-renderer";

import { CollectionCard } from "./collection-card";
import { CardType } from "../../../types/CardType";

describe("CollectionCard component", () => {
  it("should render", () => {
    const props: CardType = {
      id: "test",
      name: "test",
      imageUrl: "test",
      role: "test",
      rarity: "test",
      need: 0,
      have: 0,
      number: "test",
    };

    const tree = create(<CollectionCard {...props}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
