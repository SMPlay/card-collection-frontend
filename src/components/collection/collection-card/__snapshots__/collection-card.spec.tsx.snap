// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsPagination component should render 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="test"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="450"
      src="test"
      title="test"
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        test
      </h2>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root makeStyles-metaText-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Номер карты: 
      test
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Роль: 
      test
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Редкость: 
      test
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Есть карт: 
      0
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Нужно карт: 
      0
    </h5>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;

exports[`CollectionCard component should render 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="test"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      height="450"
      src="test"
      title="test"
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        test
      </h2>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root makeStyles-metaText-1"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Номер карты: 
      test
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Роль: 
      test
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Редкость: 
      test
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Есть карт: 
      0
    </h5>
    <h5
      className="MuiTypography-root MuiTypography-h6"
    >
      Нужно карт: 
      0
    </h5>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
